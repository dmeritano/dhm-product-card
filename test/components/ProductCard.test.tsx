import * as React from 'react';
import renderer from "react-test-renderer"
import { ProductCard } from "../../src/components"
import { product1 } from '../data/products';

const { act } = renderer

describe("ProductCard", () => {

    test("Should show the component correctly", () => {

        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <h1>Product Card Test</h1>
                    )
                }
            </ProductCard>
        )

        expect ( wrapper.toJSON()).toMatchSnapshot()
    })

    test("Should increase the counter", () => {

        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    ( {count, increaseBy }) => (
                        <>
                            <h1>Product Card Test</h1>
                            <span>{count}</span>
                            <button onClick={ () => increaseBy(1) }>+1</button>
                        </>
                    )
                }
            </ProductCard>
        )

        let tree:any = wrapper.toJSON()
        expect (tree ).toMatchSnapshot()

        act( ()=> {
            tree.children[2].props.onClick();
        })
        
        tree = wrapper.toJSON();
        expect(tree.children[1].children[0]).toBe("1")    
    
    })

})