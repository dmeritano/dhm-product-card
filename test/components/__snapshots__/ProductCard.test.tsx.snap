// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Should increase the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card Test
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;

exports[`ProductCard Should show the component correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card Test
  </h1>
</div>
`;
